### PR TITLE
Соловьева Мария, 2536

### PR DESCRIPTION
**Pull request by s-sladen**

### CODE REVIEW

**Comment by thevery** [comment]
You've already got 8 points since last year, aren't you?

**Comment by s-sladen** [comment]
Yes. But the task was a bit different.

**Comment by thevery** [comment]
exactly the same actually :)

**Comment by s-sladen** [comment]
Last year the task was to write everything from scratch, not to optimize existing code. This is the only difference.

**Comment by thevery** [comment]
ok = 8
~30 FPS during non-looped cycles

File: app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java [hunk 1]
```diff
@@ -10,18 +11,29 @@
...
-    int height = 0;
-    int scale = 4;
-    final int MAX_COLOR = 10;
-    int[] palette = {0xFFFF0000, 0xFF800000, 0xFF808000, 0xFF008000, 0xFF00FF00, 0xFF008080, 0xFF0000FF, 0xFF000080, 0xFF800080, 0xFFFFFFFF};
-    SurfaceHolder holder;
-    Thread thread = null;
-    volatile boolean running = false;
+    private static final int MAX_COLOR = 10;
+    private static final int[] PALETTE = {0xFFFF0000, 0xFF800000, 0xFF808000, 0xFF008000, 0xFF00FF00, 0xFF008080, 0xFF0000FF, 0xFF000080, 0xFF800080, 0xFFFFFFFF};
+    private static final int SCALE = 4;
```
Comment (thevery) [inline app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java:19]: you should use 320*240 fixed size instead

File: app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java [hunk 2]
```diff
@@ -53,61 +66,77 @@ public void run() {
...
+                            field[y2][x1] == nextColor || field[y2][x] == nextColor || field[y2][x2] == nextColor) {
+                        tempField[y][x] = nextColor;
+                        colors[y * width + x] = PALETTE[nextColor];
+                    } else {
+                        tempField[y][x] = field[y][x];
+                        looped = false;
                     }
                 }
             }
+            {
```
Comment (thevery) [inline app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java:129]: extra `{}`

**Pull request closed without merging**